model/generate.go

```diff
@@ -304,7 +319,9 @@ func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, v *Version
 // CheckForCycles builds a dependency graph from the existing tasks in the version and simulates
 // adding the generated tasks, their dependencies, and dependencies on the generated tasks to the graph.
 // Returns a DependencyCycleError error if the resultant graph contains dependency cycles.
-func (g *GeneratedProject) CheckForCycles(v *Version, p *Project, projectRef *ProjectRef) error {
+func (g *GeneratedProject) CheckForCycles(ctx context.Context, v *Version, p *Project, projectRef *ProjectRef) error {
+	ctx, span := tracer.Start(ctx, "simulate-dependencies")
```

Comment:
Why "simulate-dependencies" over "Check-for-cycles"?

Reply:
That's what it was called in the old splunk logs, but you're right we should change it